domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimState.java

```diff
@@ -15,4 +15,13 @@
     public String getValue() {
         return state;
     }
+
+    public static ClaimState fromValue(String state) {
+        for (ClaimState claimState : ClaimState.values()) {
+            if (claimState.getValue().equalsIgnoreCase(state)) {
+                return claimState;
+            }
+        }
+        throw new IllegalArgumentException(state + " is not a valid state.");
```

Comment:
Something like this should be easier to read
```
return Arrays.stream(ClaimState.values())
    .filter(claimState -> claimState.getValue().equalsIgnoreCase(state))
    .findAny()
    .orElseThrow(() -> new IllegalArgumentException(state + " is not a valid state.");
```

Reply:
I felt going out of the way to create a stream out of an array would be an overkill in this case. Have a look at this https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html

If you still insist and feel this is necessary, I will change it.

Reply:
done